NSS-Day-Cohort-53/tabloid-full-stack-always-polite-iguanas
# Logged in admins can edit a category

NOTES: Commented out code in `UserprofileController.cs` was from getting ahead of myself yesterday and starting the wrong ticket. view user profiles will be my next ticket and that is the start of that controller method for it  

closes issue #36 

#### Changes Made
1. Added file `categoryEdit.js` to `src/components/Categories` directory.
1. Modified file `categoryManager.js` to include `updateCategory` to update the database.
2. Modified file `ApplicationViews.js` to include routing for the edit category view.
3. Modified file `CategoryController.cs" to alter the response of the httpPut method 
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout 36_DS_Edit_Category
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    > IF a user is logged in AND is an admin WHEN they click the edit link they are taken to a form with the category name pre filled. After the user makes their changes and clicks submit the database is updated and the user is redirected to the list of categories.

6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
